### PR TITLE
Zephyr User Mode Support

### PR DESCRIPTION
**Pull request by mxsdlr**
Add support for the [Zephyr Usermode](https://docs.zephyrproject.org/latest/kernel/usermode/index.html) to the Zephyr port.

The following changes are applied, if this PR gets accepcted:

- Fix `signbit`, check if it is defined already and only implement it, if not
- Introduce `sys_mutex` and `sys_sem` in favour of `k_mutex` and `k_sem`, when `CONFIG_USERMODE` is enabled
- Remove the installation of the `_stdout_hook_iwasm()` when `CONFIG_USERMODE` is enabled, otherwise this causes MPU errors since the std hook is in the kernel space
- Add a thread name for debugging

### CODE REVIEW

File: core/shared/platform/zephyr/platform_internal.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 /*
  * Copyright (C) 2019 Intel Corporation.  All rights reserved.
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * SPDX-FileCopyrightText: 2024 Siemens AG (For Zephyr usermode changes)
```
Comment (wenyongh) [inline core/shared/platform/zephyr/platform_internal.h:5]: If using a new license, had better add the license file under current folder, and then add the related info in `<wamr_root>/ATTRIBUTIONS.md` (refer to other components and licenses), could you help do that?

File: core/shared/platform/zephyr/zephyr_platform.c [hunk 2]
```diff
@@ -1,10 +1,13 @@
 /*
  * Copyright (C) 2019 Intel Corporation.  All rights reserved.
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * SPDX-FileCopyrightText: 2024 Siemens AG (For Zephyr usermode changes)
  */
 
 #include "platform_api_vmcore.h"
 #include "platform_api_extension.h"
+#include "bh_log.h"
```
[the remark this answers is not in the document]
Comment (mxsdlr) [reply core/shared/platform/zephyr/zephyr_platform.c]: Thank you for paying attention. I removed this line, you are right, it's not needed.

File: core/shared/platform/zephyr/platform_internal.h (hunk 1 above)
Comment (mxsdlr) [reply core/shared/platform/zephyr/platform_internal.h:5]: I'm currently working on this as I have to check for internal regulations. I will post any updates as soon as I know something. Thank you for pointing out the `ATTRIBUTIONS.md`

Comment (wenyongh) [inline core/shared/platform/zephyr/platform_internal.h:5]: @mxsdlr could you move this line to be after L2, like the files below?

**Review by wenyongh — APPROVED** [review]
LGTM

**Pull request merged**